tools/project.py

```diff
@@ -155,6 +155,12 @@ def __init__(self) -> None:
         self.custom_build_steps: Optional[Dict[str, List[Dict[str, Any]]]] = (
             None  # Custom build steps, types are ["pre-compile", "post-compile", "post-link", "post-build"]
         )
+        self.make_clangd_config: bool = (
+            False  # Generate compile_commands.json for clangd
+        )
+        self.clangd_compiler_defines: bool = (
+            False  # Automatically apply defines for __MWERKS__ and such in clangd config
+        )
```

Comment:
This feels pretty RB3-specific. How would you feel about making a `clangd_defines` config option instead? Then RB3 can implement the `COMPILER_DEFINES` logic in its `configure.py`, and it simplifies the `project.py` logic a bit.

Reply:
I figured this might be useful for other projects as well.

The benefit of doing this in the generation is that it can choose the appropriate defines and values based on the compiler version being used, rather than assuming a fixed set for the whole project, but I suppose that's more of a nicety than a necessity.

Reply:
Could you show an example of where this is useful? Just struggling to wrap my head around the purpose

Reply:
For our purpose, it's mainly ease of integration and future-proofing regarding STLport. It has configurations for a bunch of compilers, and rather than tearing out those configs and meticulously removing their associated hacks throughout the library, like I did prior to our recent downgrade of it, I opted to simply add new configurations for Wii/GC Metrowerks:

![image](https://github.com/user-attachments/assets/7cd4d19e-f6e7-4ef2-8a62-3c27ab6aea08)

Without the defines, these preprocessor paths don't get hit, and it trips STLport's "can't detect compiler" error case.

We don't *really* need to have the exact defines/values for specific compiler versions, though. I mostly just thought it could be handy for someone else, but I have no real basis on whether that would actually be the case.

Reply:
> We don't _really_ need to have the exact defines/values for specific compiler versions, though. I mostly just thought it could be handy for someone else, but I have no real basis on whether that would actually be the case.

So with a theoretical `config.clangd_flags`, could RB3 just pass in `-D__MWERKS__` and get the desired behavior?

I ask because, while a cool idea and I appreciate the testing you did to come up with the compiler defines list, it still doesn’t feel quite right imo. `__MWERKS__`, to me, is for when you’re _actually_ compiling with MWCC and are using MWCC specific features. Therefore it shouldn’t be enabled with clangd: `__clang__` is defined instead.

Maybe your stlport changes could check for an `RVL` flag (need to look into what the “official” SDK flags would be), which indicates the target _platform_ instead of the specific compiler being used.

Reply:
`config.clangd_flags` wouldn't be necessary, that's what `.clangd` is for. But yes, we definitely can just define things ourselves. I'll remove the compiler define stuff for now, and if someone else wants it down the road, we can reconsider then.

As to why we use the MWCC defines instead of something else: STLport is specifically oriented around compilers and not just platforms, as it needs to know about any potential language support and standard library issues. The already-provided `stl_mwerks.h` configuration was made for desktop MWCC, and has several things which we explicitly don't want for our configuration. We certainly could provide a more dedicated config for clangd, but since we're tied to MWCC's libraries for the time being, it seems most logical to me to act like we're compiling under MWCC regardless.